getmntent(3c) requires to include stdio.h:

GETMNTENT(3C)            Standard C Library Functions            GETMNTENT(3C)

NAME
     getmntent, getmntany, getextmntent, hasmntopt, putmntent, resetmnttab –
     get mounted device information

SYNOPSIS
     #include <stdio.h>
     #include <sys/mnttab.h>
diff -wpruN --no-dereference '--exclude=*.orig' a~/include/vki/vki-solaris.h a/include/vki/vki-solaris.h
--- a~/include/vki/vki-solaris.h	1970-01-01 00:00:00
+++ a/include/vki/vki-solaris.h	1970-01-01 00:00:00
@@ -678,6 +678,7 @@ typedef struct vki_kcf_door_arg_s {
 #define VKI_MNTIOC_GETMNTANY MNTIOC_GETMNTANY
 
 
+#include <stdio.h>
 #include <sys/mnttab.h>
 #define vki_extmnttab extmnttab
 #define vki_mntentbuf mntentbuf
